homeassistant/components/ozw/config_flow.py

```diff
@@ -1,24 +1,174 @@
 """Config flow for ozw integration."""
+import logging
+
+import voluptuous as vol
+
 from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
 
+from .const import CONF_INTEGRATION_CREATED_ADDON
 from .const import DOMAIN  # pylint:disable=unused-import
 
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDON_DEVICE = "device"
+CONF_ADDON_NETWORK_KEY = "network_key"
+CONF_NETWORK_KEY = "network_key"
+CONF_USB_PATH = "usb_path"
+CONF_USE_ADDON = "use_addon"
 TITLE = "OpenZWave"
 
+ON_SUPERVISOR_SCHEMA = vol.Schema({vol.Optional(CONF_USE_ADDON, default=False): bool})
+
 
 class DomainConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for ozw."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
 
+    def __init__(self):
+        """Set up flow instance."""
+        self.addon_config = None
+        self.network_key = None
+        self.usb_path = None
+        self.use_addon = False
+        # If we install the add-on we should uninstall it on entry remove.
+        self.integration_created_addon = False
+
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
+
+        if not self.hass.components.hassio.is_hassio():
+            return self._async_use_mqtt_integration()
+
+        return await self.async_step_on_supervisor()
+
+    @callback
+    def _async_use_mqtt_integration(self):
+        """Handle logic when using the MQTT integration."""
```

Comment:
outsider-eyes comment: the method name sorta already says what this comment says. these are some of the questions that come into my head when i see this:

- What "logic" is it handling? Is it setting something up, or just verifying it?
- "when using the MQTT integration" vs what alternate option?

maybe the comment could answer those, or some similar questions?

Reply:
This method should be the entry point to all logic that pertains to the MQTT integration. I don't want to be more specific than that to allow the details to change within that scope. I'll clarify the docstring.

Reply:
```suggestion
        """Handle logic when using the MQTT integration.
        
        This is the entry point for the logic that is needed
        when this integration will depend on the MQTT integration.
        """
```